src/qibolab/_core/instruments/qm/config/elements.py

```diff
@@ -41,19 +26,33 @@ class OutputSwitch:
     """
 
 
-def _to_port(channel: Channel) -> dict[str, tuple[str, int]]:
-    """Convert a channel to the port dictionary required for the QUA config."""
-    return {"port": (channel.device, channel.port)}
+def _to_port(channel: Channel) -> dict[Port, InOutType]:
```

Comment:
Concerning the return type hint, in this case it works fine because you only have one element. But it would not scale for a dictionary with multiple (inhomogeneous) entries.

In general, the tool to define these kinds of type hints (without promoting them to an actual dataclass), is to use a [`TypedDict`](https://docs.python.org/3/library/typing.html#typing.TypedDict).
You can define it inline (functional style) or with a dataclass-like syntax. Still it will be a hint for a plain `dict`, it does not require different runtime classes.

Reply:
Thanks for the suggestion, good to know. I updated several type hints in this file to use this in 4d00aebb3855dcb8b0dfca9f5b7050c062d2ee2f, let me know if you agree.

Reply:
Yes, exactly :)

Thanks!